pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -110,7 +120,7 @@ type Registry struct {
 	// A registry is an Endpoint too
 	Endpoint
 	// The registry's mirrors.
-	Mirrors []Endpoint `toml:"mirror,omitempty"`
+	Mirrors []Mirror `toml:"mirror,omitempty"`
```

Comment:
Hum, this is probably an API break. And putting `MirrorByDigestOnly` into `Endpoint` means there are two instances of that flag in a `Registry`.

This is going to be “fun”. (I didn’t spend time thinking how to deal with this. The nuclear option is to leave the public data types as is, and build a parallel implementation with private data types. Hopefully we can do something easier.)

Reply:
I don't think we need a new struct.  Moving `MirrorByDigestOnly bool` from `type Registry` to `type Endpoint` should do the trick.  This way, we continue supporting the top-level field of the registry and add support for mirrors as well.

Reply:
Just moving it would break existing users like https://github.com/openshift/runtime-utils/blob/8b8348d80d1d1e7b6cf06fb009d5965e0b55baa2/pkg/registries/registries.go#L162 . Go API stability rules.

Reply:
… Oh, I see what you mean; that one would work. But
```go
	v := Registry {
		MirrorByDigestOnly: true,
	}
```
```
./prog.go:50:3: cannot use promoted field Endpoint.MirrorByDigestOnly in struct literal of type Registry
```

Reply:
I think that would still work since the type is embedded, isn't it?  Only initialization wouldn't work, e.g.:
```go
foo := Registry {
      MirrorByDigestOnly: true,
}
```

(Lawyer talking): the types are meant to be read, not written, which makes me OK to do this change.

Reply:
Heh, I would argue the types are published only to be written (because an old writer can reasonably write a subset of the fields, if any new fields are added in the future), _not_ to be read (because an old reader will misinterpret configuration with unknown fields).

Reply:
An alternative would be not use the very same name in the structs and have an `Endpoint.DigestOnly`.  That would practically then also apply to use the main registry only for digests which seems like a legit use case.

Reply:
> An alternative would be not use the very same name in the structs and have an `Endpoint.DigestOnly`.

Yes.

> That would practically then also apply to use the main registry only for digests which seems like a legit use case.

That doesn’t make much sense to me. The primary registry can be unavailable, but to prohibit its use for _some_ accesses if it is available, and to prohibit especially those accesses for which that registry is the _only_ actual authority, is strange.

I’d prefer to reject such configurations, if only to avoid the confusion between the existing field (which affects all mirrors) and the new field (if it affects that single endpoint), at least until we have a clear use case.

Reply:
Sounds good to me :heavy_check_mark: